cmd/vcluster/context/context.go

```diff
@@ -162,6 +172,18 @@ func NewControllerContext(currentNamespace string, localManager ctrl.Manager, vi
 		return nil, fmt.Errorf("you cannot sync storage classes and legacy storage classes at the same time. Choose only one of them")
 	}
 
+	// enable additional controllers required for scheduling with storage
+	if options.EnableScheduler && controllers["persistentvolumeclaims"] {
+		klog.Infof("persistentvolumeclaim syncing and scheduler enabled, enabling required controllers: %q", schedulerRequiredControllers)
+		for _, ctr := range schedulerRequiredControllers {
+			controllers[ctr] = true
+		}
+		if !(controllers["storageclasses"] || controllers["legacy-storageclasses"]) {
+			klog.Info("persistentvolumeclaim syncing and scheduler enabled, but storageclass sync not enabled. Syncing host storageclasses to vcluster(legacy-storageclasses)")
+			controllers["legacy-storageclasses"] = true
```

Comment:
Oh, I wish we hadn't named it "legacy-storageclasses", because it feel odd to default to that :laughing: 
But it makes sense.

Reply:
We will actually rename it :) - #783